src/components/TodoList.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from "react";
+import uuid from "uuid";
+import Todo from "./Todo";
+import TodoInput from "./TodoInput";
+import SortTodosForm from "./SortTodosForm";
+import FilterTodosSection from "./FilterTodosSection";
+import DeleteTodosSection from "./DeleteTodosSection";
+
+export default class TodoList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      todos: [
+        { task: "have some coffee", id: uuid(), completed: false },
+        { task: "walk the dog", id: uuid(), completed: false },
+        { task: "take over the world", id: uuid(), completed: false }
+      ],
+      displayTodos: "all",
+      sortByStatus: false
+    };
+  }
+
+  createTodo = newTodo => {
+    this.setState({
+      todos: [...this.state.todos, newTodo]
+    });
+  };
+
+  updateTodo = (id, updatedTask) => {
+    const updatedTodos = this.state.todos.map(todo => {
```

Comment:
tutaj ta sama sytuacja. Ustawiasz w setState coś co przeczytałaś z this.state

Powinno być tak:
```
this.setState(prevState => {
    const updatedTodos = prevState.todos.map(todo => {
    if (todo.id === id) {
        return { ...todo, task: updatedTask };
      } else {
        return todo;
      }

     return updatedTodos;

})
```